### DISCUSSION TITLE
Remove all groups synced from LDAP

### DISCUSSION THREAD

**Discussion by GereG13** [keycloak/keycloak#16682] [Q&A]
After entering **LDAP Groups DN** field under **User Federation** => **Ldap** => **LDAP Mappers** => **groups** , and doing **Sync LDAP Groups to Keycloak** , I realized more than 1000 groups got downloaded.

I tried updating the **LDAP Filter** field (as shown in below screenshot) and ran the Sync again but all the 1000 groups are still there instead of Keycloak only keeping the Filtered ones and removing all other groups.

How to remove all groups to start clean or make Keycloak only keep the groups based on the **LDAP Filter** used?

When I added the **LDAP Filter** I was expecting the have Keycloak remove all other groups synced before and only keeping the ones matching the filter, but instead it kept them all.

![image](https://user-images.githubusercontent.com/106213200/215181574-8f1b7cc9-1fe2-4621-bcba-08a85c874eac.png)

**Comment by avhb** [comment]
I am facing kind of a similar issue in which that when I delete the entire ldap integration, keycloak prompts the admin that all related data will be removed, but that is not the case for groups synced from ldap, they stay even though all users inside those groups are deleted
